### PR TITLE
use addHandlerForQuerySelector

### PR DESCRIPTION
**Pull request by labkey-matthewb**
#### Rationale
use addHandlerForQuerySelector when rendering helpPopup to reduce number or registered events.
48714

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

File: api/src/org/labkey/api/util/PageFlowUtil.java [hunk 1]
```diff
@@ -1611,6 +1612,10 @@ public static HelpPopupBuilder popupHelp(@NotNull HtmlString helpHtml, String ti
     }
 
 
+    // NOTE: these are attached vial query selector, so they must be the same for every popup
```
Comment (labkey-martyp) [inline api/src/org/labkey/api/util/PageFlowUtil.java]:
```suggestion
    // NOTE: these are attached via query selector, so they must be the same for every popup
```

**Review by labkey-martyp — APPROVED** [review]
Looks good. I did a quick test to make sure the popups were still working. And this cut the event handler count way down.

**Comment by labkey-jeckels** [comment]
Would it be worthwhile to cap the number of handlers in `PageConfig` to something that's ridiculously high but enough to protect the server? 100,000? 1,000,000? We could log an error/warning.

**Comment by labkey-matthewb** [comment]
> Would it be worthwhile to cap the number of handlers in `PageConfig` to something that's ridiculously high but enough to protect the server? 100,000? 1,000,000? We could log an error/warning.

Probably a good idea.

**Pull request merged**